src/gui/editorwidgets/qgslistwidgetwrapper.cpp

```diff
@@ -70,16 +70,32 @@ QVariant QgsListWidgetWrapper::value() const
 {
   QVariant::Type type = field().type();
   if ( !mWidget ) return QVariant( type );
+  const QVariantList list = mWidget->list();
   if ( type == QVariant::StringList )
   {
-    QStringList result;
-    const QVariantList list = mWidget->list();
-    for ( QVariantList::const_iterator it = list.constBegin(); it != list.constEnd(); ++it )
-      result.append( it->toString() );
-    return result;
+    if ( list.size() == 0 && config( QStringLiteral( "EmptyIsNull" ) ).toBool() )
```

Comment:
This `if` statement and block looks to be duplicated on L:90-93 below. If that is intentional, I think it could be de-duplicated and pulled out to be a "pre-check" outside of the current containing if statement (i.e. it can be inserted in-between L:73-74 above), which would help reduce the nesting and make the control flow a bit easier to follow.

Reply:
yup, good point, 👍